### PR TITLE
Fix rsyslogd start failed cause by rsyslog.conf is emtpy.

### PR DESCRIPTION
**Pull request by lerry-lee**
Signed-off-by: Chun'ang Li <chunangli@microsoft.com>

#### Why I did it

In to-sonic and multi-asic KVM-test, pretest sometimes failed. **Reason is rsyslogd process can not start in teamd container**. Because **rsyslog.conf is empty** caused by **sonic-cfggen execute failed**, and failed log is:

image

#### How I did it

If `sonic-cfggen -d` execute failed, execute without `-d` because the template file has the default value.

#### How to verify it

Build image and test it over 40 times, all passed pretest.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211

### CODE REVIEW

**Review by yejianquan — APPROVED** [review]
LGTM

**Comment by saiarcot895** [comment]
Is it known why sonic-cfggen fails?

**Comment by Junchao-Mellanox** [comment]
Should we fix sonic-cfggen? It should not generate an empty file while exception occurs?

**Comment by yejianquan** [comment]
> Is it known why sonic-cfggen fails?
> 
> Edit: never mind, I just saw the screenshot.

@saiarcot895 
Because sonic-cfggen can't connect to the redis.
I pulled you into an email thread that contains more technical detail.

**Comment by lerry-lee** [comment]
> Should we fix sonic-cfggen? It should not generate an empty file while exception occurs?

@Junchao-Mellanox 
Thanks for the suggestion! But it's a hotfix, we think should resolve the problem first, because it has affected some PR already. For deep reason and sonic-cffgen improvement, we will follow up continuously.

File: files/build_templates/docker_image_ctl.j2 [hunk 1]
```diff
@@ -44,12 +44,19 @@ function updateSyslogConf()
...
     sonic-cfggen -d -t /usr/share/sonic/templates/rsyslog-container.conf.j2 -a "{\"target_ip\": \"$TARGET_IP\", \"container_name\": \"$CONTAINER_NAME\" }"  > $TMP_FILE
+    if [ $? -ne 0 ]; then
+      echo "Error: Execute sonic-cfggen failed."
+      failed="true"
+    fi
     {%- endif %}
-    docker cp $TMP_FILE ${DOCKERNAME}:/etc/rsyslog.conf
+    if [ "$failed" == "false" ]; then
+      docker cp $TMP_FILE ${DOCKERNAME}:/etc/rsyslog.conf
+    fi
```
Comment (wangxin) [inline files/build_templates/docker_image_ctl.j2:59]: There is a risk that teamd container started without proper rsyslog.conf file. Then probably the log of teamd container can't be received by the host rsyslog service. The consequence is that teamd syslog could be missing.

Comment (lerry-lee) [reply files/build_templates/docker_image_ctl.j2:59]: Thanks for the reminder! Will discuss and improve again.

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by wangxin — APPROVED** [review]

**Comment by lerry-lee** [comment]
https://github.com/sonic-net/sonic-buildimage/issues/13674 will trace this issue.

**Comment by StormLiangMS** [comment]
@yejianquan  @wangxin do we need this for 202205?

**Pull request merged**